### PR TITLE
fix: dns sdk fix

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
Fix dns sdk. 
- (1)Add missing field.
- (2)Mark `Deprecated` to invalid field.
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format,
  will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: openstack/dns/v2/recordsets/results.go [hunk 1]
```diff
@@ -93,12 +93,16 @@ type RecordSet struct {
 	// Status is the status of the recordset.
 	Status string `json:"status"`
 
+	Default bool `json:"default"`
+
+	// Deprecated
```
Comment (ShiChangkuo) [inline openstack/dns/v2/recordsets/results.go:98]: please put the deprecated fields together.

File: openstack/dns/v2/recordsets/results.go [hunk 2]
```diff
@@ -111,16 +115,20 @@ type RecordSet struct {
 	// Links includes HTTP references to the itself,
 	// useful for passing along to other APIs that might want a recordset
 	// reference.
-	Links []golangsdk.Link `json:"-"`
+	Links *LinkResult `json:"links"`
```
Comment (ShiChangkuo) [inline openstack/dns/v2/recordsets/results.go:118]: please revert this, we need to reanalyze this.

File: openstack/dns/v2/recordsets/results.go (hunk 1 above)
Comment (deer-hang) [reply openstack/dns/v2/recordsets/results.go]: Ok, Please check it again.

File: openstack/dns/v2/recordsets/results.go (hunk 2 above)
Comment (deer-hang) [reply openstack/dns/v2/recordsets/results.go]: Done.

**Comment by deer-hang** [comment]
I have run the test of resource `huaweicloud_dns_zone` and resource `huaweicloud_dns_recordset` with the changed golangsdk.
The result is currect.

**Pull request merged**